wallet/src/main/java/com/android/identity_credential/wallet/presentation/PresentmentFlow.kt

```diff
@@ -0,0 +1,350 @@
+import com.android.identity.cbor.Cbor
+import com.android.identity.crypto.Algorithm
+import com.android.identity.crypto.javaX509Certificates
+import com.android.identity.document.Document
+import com.android.identity.document.DocumentRequest
+import com.android.identity.document.NameSpacedData
+import com.android.identity.issuance.CredentialFormat
+import com.android.identity.issuance.DocumentExtensions.documentConfiguration
+import com.android.identity.mdoc.credential.MdocCredential
+import com.android.identity.mdoc.mso.MobileSecurityObjectParser
+import com.android.identity.mdoc.mso.StaticAuthDataParser
+import com.android.identity.mdoc.request.DeviceRequestParser
+import com.android.identity.mdoc.response.DeviceResponseGenerator
+import com.android.identity.mdoc.response.DocumentGenerator
+import com.android.identity.mdoc.util.MdocUtil
+import com.android.identity.securearea.KeyLockedException
+import com.android.identity.securearea.KeyUnlockData
+import com.android.identity.securearea.software.SoftwareKeyInfo
+import com.android.identity.securearea.software.SoftwareKeyUnlockData
+import com.android.identity.securearea.software.SoftwareSecureArea
+import com.android.identity.trustmanagement.TrustPoint
+import com.android.identity.util.Constants
+import com.android.identity.util.Logger
+import com.android.identity_credential.wallet.R
+import com.android.identity_credential.wallet.WalletApplication
+import com.android.identity_credential.wallet.ui.prompt.biometric.showBiometricPrompt
+import com.android.identity_credential.wallet.ui.prompt.consent.showConsentPrompt
+import com.android.identity_credential.wallet.ui.prompt.passphrase.showPassphrasePrompt
+import kotlinx.datetime.Clock
+
+/**
+ * Class responsible for showing the Presentment Flow for every credential [Document] that is
+ * requested by some "Verifying party". The purpose of the Presentment Flow is to authorize a
+ * [Document] to be "sent" to the Verifier by showing a series of consent and authentication prompts
+ * to the user. After a successful Presentment, the [Document] used during the Presentment is added
+ * to the response to the Verifier. When all Presentment Flows have ended (successfully
+ * or not), the response bytes are generated and contain all the [Document]s from successful
+ * Presentments.
+ *
+ * An [Exception] (mostly an [IllegalStateException]) is thrown for numerous reasons, such as
+ * if credentials could not be found in a Document; authentication type was not specified when
+ * showing Biometric prompt; a (new) secure area has not been handled/implemented yet;
+ * if a prompt is unsuccessful because the user cancelled or was not able to authenticate, etc..
+ *
+ * @param walletApp the WalletApplication instance used for dependencies.
+ * @param fragmentActivity used for showing Dialog Fragments.
+ */
+class PresentmentFlow(
+    private val walletApp: WalletApplication,
+    private val fragmentActivity: FragmentActivity
+) {
```

Comment:
Why is this a class and now just a single function? I mean, I expected it to look like this:

```
    suspend fun showPresentmentFlow(
        activity: FragmentActivity,
        walletApp: WalletApplication,
        documentRequest: DocumentRequest,
        encodedSessionTranscript: ByteArray
    ): ByteArray
```

Also - as I alluded to in the above - this needs to take a `DocumentRequest` instead of DeviceRequest CBOR because we want it to be usable also for Credman and OpenID4VP and the request format in both of those cases are not CBOR.

Also, it should not handle multiple document requests in a single request, that's the wrong layer at the stack to do this. That's only relevant for proximity presentations (ie. when you have DeviceRequest CBOR) and in this case the caller (e.g. `PresentationActivity`) can simply just parse the CBOR themselves, get all the N `DocumentRequest` instances, and then call `showPresentmentFlow()` for each of them.

Also, btw, the longer-term goal is to factor this function - as well as things like `DocumentModel` and `showBiometricPrompt()` into a `identity-appsupport` library which `wallet` and also things like `wallet-wearos` can call into. So it can't really take a `WalletApplication` and thus we need at some point to feed the dependencies directly... but no need to change that right now, should be an easy change once we get there...